### ISSUE TITLE
PetSet fails to update on replica count change

### ISSUE THREAD

**Issue by etki** [kubernetes/kubernetes#30453]
I'm getting "updates to petset spec for fields other than 'replicas' are forbidden." error when changing number of replicas in petset configuration. 

This happens when i apply following dummy configuration

```
apiVersion: apps/v1alpha1
kind: PetSet
metadata:
  name: dummy-petset
spec:
  serviceName: nginx
  replicas: 2
  template:
    metadata:
      labels:
        app: nginx
      annotations:
        pod.alpha.kubernetes.io/initialized: "true"
    spec:
      containers:
      - name: nginx
        image: nginx
```

and then put any number of replicas other than 2:

```
kubectl apply -f petset.yml 
petset "dummy-petset" created

kubectl apply -f petset.yml 
proto: no encoder for TypeMeta unversioned.TypeMeta [GetProperties]
proto: tag has too few fields: "-"
proto: no coders for struct *reflect.rtype
proto: no encoder for sec int64 [GetProperties]
proto: no encoder for nsec int32 [GetProperties]
proto: no encoder for loc *time.Location [GetProperties]
proto: no encoder for Time time.Time [GetProperties]
proto: no encoder for i resource.int64Amount [GetProperties]
proto: no encoder for d resource.infDecAmount [GetProperties]
proto: no encoder for s string [GetProperties]
proto: no encoder for Format resource.Format [GetProperties]
proto: no encoder for InitContainers []v1.Container [GetProperties]
proto: no coders for intstr.Type
proto: no encoder for Type intstr.Type [GetProperties]
proto: no encoder for TypeMeta unversioned.TypeMeta [GetProperties]
The PetSet "dummy-petset" is invalid.
spec: Forbidden: updates to petset spec for fields other than 'replicas' are forbidden.
```

version output:
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:29:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:22:25Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

This occurs on freshly-created GKE cluster and i haven't touched any internals, of course.

**Comment by bprashanth** [comment]
You can't apply in alpha petset because of: https://github.com/kubernetes/kubernetes/issues/30158
As the docs indicate, you need to edit or patch: https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/user-guide/petset.md#updating-a-pet-set
This is fixed in 1.4.

**Comment by etki** [comment]
@bprashanth got it, thank you
